src/index.js

```diff
@@ -0,0 +1,140 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const increaseTempButton = document.getElementById('increaseTempControl');
+    const decreaseTempButton = document.getElementById('decreaseTempControl');
+    const landscape = document.getElementById('landscape');
+    const cityDisplay = document.getElementById('headerCityName');
+    const cityInput = document.getElementById('cityNameInput');
+    const getTempButton = document.getElementById('currentTempButton');
+    const skySelect = document.getElementById("skySelect");
+    const sky = document.getElementById("sky");
+    const resetButton = document.getElementById("cityNameReset");
+    const DEFAULT_TEMP = 70;
+
+    cityDisplay.textContent = "Miami";
+    cityInput.addEventListener('input', () => {
+      cityDisplay.textContent = cityInput.value.trim();
+    });
+
+    landscape.style.fontFamily = "monospace";
+    landscape.style.whiteSpace = "pre";
+
+    let temperature = DEFAULT_TEMP;
+    const PROXY_URL = "http://localhost:5000";
+
+    const getRealTimeTemperature = async () => {
+      const city = cityDisplay.textContent.trim();
+  
+      if (!city) {
+        alert("Please enter a valid city name.");
+        return;
+      }
+
+      try {
+        const locationResponse = await axios.get(`${PROXY_URL}/location`, {
+          params: { q: city }
+        });
+
+        if (!locationResponse.data || locationResponse.data.length === 0) {
+          alert('Unable to find location.');
+          return;
+        }
+
+        const { lat, lon } = locationResponse.data[0];
+        const weatherResponse = await axios.get(`${PROXY_URL}/weather`, {
+          params: { lat: lat, lon: lon }
+        });
+
+        if (weatherResponse.data && weatherResponse.data.main) {
+       
+          const kelvinTemp = weatherResponse.data.main.temp;
+          temperature = Math.round((kelvinTemp - 273.15) * 9/5 + 32);
+          updateDisplay();
+        } else {
+          alert('Unable to fetch weather data.');
+        }
+      } catch (error) {
+        console.error('Error fetching data:', error.message);
+        alert('Unable to fetch temperature. Please try again.');
+      }
+    };
```

Comment:
This function is getting long, how could we divide up the different steps we take into helper function to better stick to the single responsibility principle and have more focused, easier to read functions?